tests/eko/io/test_legacy.py

```diff
@@ -14,51 +14,48 @@
 from numpy.testing import assert_almost_equal
 from ekomark.benchmark.runner import Runner
 from ekomark.benchmark.external import LHA_utils
+import eko.basis_rotation as br
 
 TEST_DATA_DIR = pathlib.Path(__file__).parents[2] / "data"   # directory of the EKO object
 pdf = toy.mkPDF("",0)
 
 x_grid = LHA_utils.toy_xgrid
-EP = (10000, 4)
+EP = (10000., 4)
 
 
 def test_read_legacy():
     for name in ["v0.13.tar", "v0.14.tar", "v0.0.tar"]:
         with eko.EKO.read(TEST_DATA_DIR / name) as evolution_operator:
-            th = evolution_operator.theory_card
-            op = evolution_operator.operator_card
             assert isinstance(
-                th, eko.io.runcards.TheoryCard
+                evolution_operator.theory_card, eko.io.runcards.TheoryCard
             )  # Check that theory card is read as theory card
             assert isinstance(
-                op, eko.io.runcards.OperatorCard
+                evolution_operator.operator_card, eko.io.runcards.OperatorCard
             )  # Check that operator card is read as operator card
 
             
             # Check if the operator has the correct dimensions
             with evolution_operator.operator(EP) as op:
-                if op.operator.shape != (14, 60, 14, 60):
-                    print("Operator does not have the correct dimensions")
+                assert op.operator.shape == (14, 60, 14, 60)
+                    
 
             # Use the operator on the pdf
             evolved_pdfs, _integration_errors = apply_pdf(
                 evolution_operator, pdf, x_grid
             )
 
         # Import the values of the LHA benchmark tables
-        lha_path = (
-            pathlib.Path(__file__).parents[4]
-            / "eko/src/ekomark/benchmark/external/LHA.yaml"
+        lha_benchmark = LHA_utils.compute_LHA_data(
+            {"FNS": "FFNS", "PTO": 0, "XIF": 1},
+            {"polarized": 0, "mugrid": 100}
```

Comment:
```suggestion
            {"polarized": False, "mugrid": 100.}
```
even if we fake, let's stick to the correct types